packages/router/src/grpc/offscreen-client.ts

```diff
@@ -15,59 +15,70 @@ const OFFSCREEN_DOCUMENT_PATH = '/offscreen.html';
 let active = 0;
 
 const activateOffscreen = async () => {
-  const offscreenExists = await chrome.runtime
+  const noOffscreen = chrome.runtime
     .getContexts({
       contextTypes: [chrome.runtime.ContextType.OFFSCREEN_DOCUMENT],
     })
-    .then(contexts => contexts.length > 0);
+    .then(offscreenContexts => !offscreenContexts.length);
 
-  if (!active || !offscreenExists) {
-    await chrome.offscreen.createDocument({
-      url: chrome.runtime.getURL(OFFSCREEN_DOCUMENT_PATH),
-      reasons: [chrome.offscreen.Reason.WORKERS],
-      justification: 'Manages Penumbra transaction WASM workers',
-    });
-  }
-
-  active++;
+  if (!active++ || (await noOffscreen))
+    await chrome.offscreen
+      .createDocument({
+        url: chrome.runtime.getURL(OFFSCREEN_DOCUMENT_PATH),
+        reasons: [chrome.offscreen.Reason.WORKERS],
+        justification: 'Manages Penumbra transaction WASM workers',
+      })
+      .catch((e: unknown) => {
+        // the offscreen window might have been created since we checked
+        // TODO: other failures?
+        console.warn('Failed to create offscreen window', e);
+      });
 };
 
+/**
+ * Decrement and close if there is no remaining activity.
+ */
 const releaseOffscreen = async () => {
-  active--;
-  if (!active) await chrome.offscreen.closeDocument();
+  if (!--active) await chrome.offscreen.closeDocument();
```

Comment:
Same comment here — no need to pack too much into an `if` statement — these can be separate lines to aid human-readability.